### PR TITLE
Added preliminary test suite. This resolves #13

### LINKED ISSUE #13: Proposal: Add a test framework
One of the requirements for us to have a mature Continuous Integration pipeline is for a test suite to be run automatically as part of the development process. Of course, before that can happen, we need a test framework.

Reading on Continuous Integration:
https://www.thoughtworks.com/continuous-integration

Popular Test Framework choices:
Jasmine:
https://www.npmjs.com/package/jasmine

Mocha:
https://www.npmjs.com/package/mocha

Benefits of mocha:
Most popular NPM test framework. There will be a billion examples for anything you could possibly want to do with mocha.

Benefits of jasmine:
A more swiss-army knife approach to a test framework than mocha. Mocha is normally used in conjunction with a BDD and a mocking library (for example: chai and sinon). With jasmine, its not necessary to pull other libraries for the BDD and mocking behavior; Its built in.

I've used both frameworks, and personally I remember liking jasmine more. Would like to hear everyone else's thoughts.

### PR DESCRIPTION
**Pull request by matsprehn**

### CODE REVIEW

File: src/randomThings.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+let user;
+
+const init = (client) => {
+	user = client.user;
+};
+const randomString = function(array, username = "", setActivity = true) {
+	if (setActivity) setRandomActivity();
+
+	const msg = array[Math.floor(Math.random() * array.length)];
+	if (msg.includes("%s")) return util.format(msg, username);
```
Comment (abrenner) [inline src/randomThings.js]: util not defined in this file causing program to crash.

File: auto-deploy/deploy-haku.sh [hunk 2]
```diff
@@ -30,6 +30,8 @@ for file in ${required_source_files}; do
 	fi
 done
 
+transfer_file_list+=" src"
+
```
Comment (abrenner) [inline auto-deploy/deploy-haku.sh]: Github will not allow me to make comments on other parts of the code, but you need to add a -r to the scp command on line 39.

```
scp -i ${deployment_key} -r ${transfer_file_list} ${deployment_user}@${deployment_host}:${deployment_path}
```

In the future I will clean up deploy scripts to make it less dependent on file placement.

File: src/randomThings.js (hunk 1 above)
Comment (matsprehn) [reply src/randomThings.js]: good catch!

File: auto-deploy/deploy-haku.sh [hunk 3]
```diff
@@ -30,11 +30,13 @@ for file in ${required_source_files}; do
...
 echo ${transfer_file_list}
 ssh -i ${deployment_key} ${deployment_user}@${deployment_host} << EOF
 [ -d ${deployment_path} ] && echo "skipping: ${deployment_path} exists" || mkdir -v ${deployment_path}
 EOF
-scp -i ${deployment_key} ${transfer_file_list} ${deployment_user}@${deployment_host}:${deployment_path}
+scp -ir ${deployment_key} ${transfer_file_list} ${deployment_user}@${deployment_host}:${deployment_path}
```
Comment (abrenner) [inline auto-deploy/deploy-haku.sh]: The is not valid syntax, see my previous comment on requirements to scp.

-i requires input which is the deployment_key. When you pass in r after i, "i" thinks that "r" the deployment key. This is a common issue when programs use the getopt function from C.

Comment (matsprehn) [reply auto-deploy/deploy-haku.sh]: TIL

**Comment by abrenner** [comment]
Looks good to me, great job @matsprehn

**Pull request merged**